drivers/edac/edac_ibecc.c

```diff
@@ -0,0 +1,454 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/pcie/pcie.h>
+
+#include <edac.h>
+#include "ibecc.h"
+
+/**
+ * In the driver 64 bit registers are used and for debug purposes we have to
+ * use printk here since LOG subsystem cannot print those values
+ */
+#include <logging/log.h>
+LOG_MODULE_REGISTER(edac_ibecc, CONFIG_EDAC_LOG_LEVEL);
+
+struct ibecc_data {
+	mem_addr_t mchbar;
+	edac_notify_callback_t cb;
+
+	/* Error count */
+	unsigned int errors_cor;
+	unsigned int errors_uc;
+};
+
+static bool ibecc_enabled(const pcie_bdf_t bdf)
+{
+	return !!(pcie_conf_read(bdf, CAPID0_C_REG) & CAPID0_C_IBECC_ENABLED);
+}
+
+static void ibecc_errcmd_setup(const pcie_bdf_t bdf, bool enable)
+{
+	uint32_t errcmd;
+
+	errcmd = pcie_conf_read(bdf, ERRCMD_REG);
+
+	if (enable) {
+		errcmd |= (ERRCMD_IBECC_COR | ERRCMD_IBECC_UC) << 16;
+	} else {
+		errcmd &= ~(ERRCMD_IBECC_COR | ERRCMD_IBECC_UC) << 16;
+	}
+
+	pcie_conf_write(bdf, ERRCMD_REG, errcmd);
+}
+
+static void ibecc_errsts_clear(const pcie_bdf_t bdf)
+{
+	uint32_t errsts;
+
+	errsts = pcie_conf_read(bdf, ERRSTS_REG);
+
+	if (!(errsts & (ERRSTS_IBECC_COR | ERRSTS_IBECC_UC))) {
+		return;
+	}
+
+	pcie_conf_write(bdf, ERRSTS_REG, errsts);
+}
+
+static const char *get_ddr_type(uint8_t type)
+{
+	switch (type) {
+	case 0:
+		return "DDR4";
+	case 3:
+		return "LPDDR4";
+	default:
+		return "Unknown";
+	}
+}
+
+static const char *get_dimm_width(uint8_t type)
+{
+	switch (type) {
+	case 0:
+		return "X8";
+	case 1:
+		return "X16";
+	case 2:
+		return "X32";
+	default:
+		return "Unknown";
+	}
+}
+
+static void mchbar_regs_dump(const struct device *dev)
+{
+	struct ibecc_data *data = dev->data;
+	uint32_t mad_inter_chan, chan_hash;
+
+	/* Memory configuration */
+
+	chan_hash =  sys_read32(HOST_REG(data->mchbar, CHANNEL_HASH));
+	printk("Channel hash %x\n", chan_hash);
```

Comment:
I guess you should be using LOG_* when you even declare your own log module in the beginning of this file?

Reply:
I replaced all printk to LOG_DBG()